### ISSUE TITLE
RawAccel config importation 

### ISSUE THREAD

**Issue by owosoup** [AndyFilter/YeetMouse#14]
The most important thing for people to migrate to this program easily would be to provide a way to replicate the exact feel it had on previous accel software. Most of the time it's RawAccel. I'm still not happy with the feeling YeetMouse has, it's not what I'm used to. I can't manage to replicate RawAccel's acceleration profile. 

If you coud automate this, by "translating" the settings.json file I have, I'd finally be able to use Linux as my main OS :D 

It would just be about copying the values and inputting them in other fields corresponding to the types of acceleration used. I'm willing to document RawAccel's features if you need info. It doesn't need FULL feature parity, but basic profile importation would help tons of people I'm sure.

**Comment by owosoup** [comment]
Side note : feature parity (not full, just the main features and user requests) comes first of course. This is for later. I'll reopen once this is relevant

**Comment by owosoup** [comment]
Now that anisotropy and angle snapping are done, I think it's time to make a basic way to convert a rawaccel configuration file, settings.json, to Yeetmouse's config.h.

**Comment by AndyFilter** [comment]
In the current state of the program I don't think it's really as basic as it seems to be. YeetMouse is still incomplete when it comes to feature parity with RawAccel, and to be honest I don't really plan to bring everything from RawAccel.
Parsing the Json from RawAccel to extract the necessary values is something that will take time, but is trivial. The real issue here, is that YeetMouse simply does not support some of the modes or functions that RawAccel does. Yes, the basic modes should work exactly the same, there is only one way to make a linear acceleration, but once you start to get into things like `motivity`, the issue arises, there are infinitely many ways of "making" a `sigmoid` function. What I used in YeetMouse is far from what RawAccel uses. It would be disingenuous to just "convert" the values from one to the other; they would feel different, even if just a bit.
The other thing is stuff like anisotropy, I may support it, but again it's done differently. In RawAccel you have a domain range, and you have the added freedom of using a specific Lp norm, which calculates velocity values a bit differently depending on the `p` value You use.
In the end, I don't think this is a good idea, at least for now. I plan to add a custom curve creation in the future (but still, there are some things of higher priority), this feature would allow to get any curve (assuming a finite precision), but then, converting RawAccel's function parameters would not only be much harder, but would also yield inaccurate results.
I'm also not a fan of having a feature that simply doesn't work on half the inputs (modes like motivity, natural etc.). Especially because most people use the more sophisticated modes rather than a linear function (correct me if I'm wrong though).

**Comment by owosoup** [comment]
You're right. My thought process didn't go that far. My request was because I want things to be easy - maybe you don't need an automatic, flawed, conversion of settings. A guide to getting a certain feel (a tutoria) can help most. Sorry for making write so much to get me to understand the problem, I get it now. Closing this since it's superfluous

**Comment by AndyFilter** [comment]
Adding some kind of tutorial / guide to the documentation about RawAccel's settings conversion is actually a pretty good idea, especially because I plan to rewrite (most of) the documentation anyway. I will try to implement something like this into the documentation with the next update (should be this month). If you have any suggestions how this section could look, feel free to share it, it will help me greatly.

**Comment by owosoup** [comment]
I'll think about it. Should I make an issue for people to suggest changes to the documentation?

**Comment by AndyFilter** [comment]
Interesting suggestion, as I would like to get some more opinions / views on different things, especially regarding the documentation, and naming. But I think for now I'll do what I think is helpful, and see how well that does.
